test/DynamicSale.test.ts

```diff
@@ -28,11 +28,13 @@ const setSigner = (user: User, context: Context) => ({
   address: user.address,
   buy: (amount: string) => ({
     success: async () => {
+      await context.deployer.TDFToken.mint(context.deployer.address, parseEther('10000'));
```

Comment:
I wouldn't recommend doing the setup in this location.
The purpose of these methods is to thoroughly examine a single action. The minting process is a part of the overall context setup for the tests.

Reply:
To add to that. The minting is done in the sale contract.
```suggestion
```